src/cpp/fastdds/xtypes/type_representation/BuiltinAnnotations.h

```diff
@@ -0,0 +1,258 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file BuiltinAnnotations.h
+ * This header file contains the declaration of the described types in the IDL file.
+ *
+ * This file was generated by the tool fastddsgen.
+ */
+
+#ifndef _FAST_DDS_GENERATED_BUILTINANNOTATIONS_H_
+#define _FAST_DDS_GENERATED_BUILTINANNOTATIONS_H_
+
+#include <array>
+#include <bitset>
+#include <cstdint>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <fastcdr/cdr/fixed_size_string.hpp>
+#include <fastcdr/xcdr/external.hpp>
+#include <fastcdr/xcdr/optional.hpp>
+
+
+
+#if defined(_WIN32)
+#if defined(EPROSIMA_USER_DLL_EXPORT)
+#define eProsima_user_DllExport __declspec( dllexport )
+#else
+#define eProsima_user_DllExport
+#endif  // EPROSIMA_USER_DLL_EXPORT
+#else
+#define eProsima_user_DllExport
+#endif  // _WIN32
+
+#if defined(_WIN32)
+#if defined(EPROSIMA_USER_DLL_EXPORT)
+#if defined(BUILTINANNOTATIONS_SOURCE)
+#define BUILTINANNOTATIONS_DllAPI __declspec( dllexport )
+#else
+#define BUILTINANNOTATIONS_DllAPI __declspec( dllimport )
+#endif // BUILTINANNOTATIONS_SOURCE
+#else
+#define BUILTINANNOTATIONS_DllAPI
+#endif  // EPROSIMA_USER_DLL_EXPORT
+#else
+#define BUILTINANNOTATIONS_DllAPI
+#endif // _WIN32
+
+namespace eprosima {
+namespace fastcdr {
+class Cdr;
+class CdrSizeCalculator;
+} // namespace fastcdr
+} // namespace eprosima
+
+
+
+namespace id {
+
+
+
+} // namespace id
+namespace autoid {
+    /*!
+     * @brief This class represents the enumeration AutoidKind defined by the user in the IDL file.
+     * @ingroup BuiltinAnnotations
+     */
+    enum AutoidKind : uint32_t
+    {
+        SEQUENTIAL,
+        HASH
+    };
+
+
+
+} // namespace autoid
+namespace optional {
+
+
+
+} // namespace optional
+namespace position {
+
+
+
+} // namespace position
+namespace extensibility {
+    /*!
+     * @brief This class represents the enumeration ExtensibilityKind defined by the user in the IDL file.
+     * @ingroup BuiltinAnnotations
+     */
+    enum ExtensibilityKind : uint32_t
+    {
+        FINAL,
+        APPENDABLE,
+        MUTABLE
+    };
+
+
+
+} // namespace extensibility
+namespace final {
+
+
+
+} // namespace final
+namespace appendable {
+
+
+
+} // namespace appendable
+/* TODO(jlbueno): Fix in Fast DDS-Gen
```

Comment:
Was this left intentionaly?

Reply:
Yes. This is an autogenerated file, so that commented lines should be fixed in Fast DDS-Gen so the empty namespaces are not generated. Not yet fixed. When fixed, the generated code would be regenerated and thus that comments would disappear.